### ISSUE TITLE
Simplify logging

### ISSUE THREAD

**Issue by ehmatthes** [ehmatthes/django-simple-deploy#260]
Originally, I didn't want to start writing a log file until we knew that the deployment was going to work. In retrospect, this was an unnecessary complexity. We're taking significant steps such as creating a project, and creating a db before we start logging. These are really important things to have a record of, especially for troubleshooting.

It's perfectly reasonable to create a log file as soon as we have something to log. People can pass the `--no-logging` arg if they need, and they can delete log files or directories before making a commit to their project if needed.

- [x] Begin logging immediately during the inspection and configuration process.
- [x] Make sure we're not logging anything sensitive such as db passwords.
    - [x] Platform.sh
    - [x] Fly.io
    - [x] Heroku
- [x] Update tests
- [x] Tests pass on Windows
    - [x] unit tests
    - [x] integration tests
- [x] Update documentation (nothing to update)
- [x] Merge this branch.
- [x] Set up PyPI 2FA.
- [x] Make new release
- [ ] Run integration test against PyPI version.
    - [x] fly_io
    - [ ] platform_sh (worked manually, as an end user would, but CLI having authentication issues from script.) See #265
    - [x] heroku
- [x] Move any significant open tasks to another issue, or roadmap doc.

**Comment by ehmatthes** [comment]
Working notes
---

- [ ] Run `build_dev_env.py`, create fresh dev env. Use `sys.exit()` calls to ensure each step is being logged appropriately.
- [ ] Create log file immediately.
- [ ] Log early steps that weren't previously being logged.
- [ ] Update conditionals in later logging steps.

**Comment by ehmatthes** [comment]
`git status` notes
---

- If there are changes that haven't been added, git status is not clean, but git diff shows no output.
- _check_git_diff()` may need output of `git status`.
    - If git diff has no output, should only find simple_deploy_logs/ in output?

**Comment by ehmatthes** [comment]
More tasks
---

- [ ] Deploy to fly_io without cli installed; does `_validate_cli()` work?

**Comment by ehmatthes** [comment]
prep_automate_all()
---

- [ ] platform_sh
- [ ] fly_io
- [ ] heroku

**Comment by ehmatthes** [comment]
Final changes before testing
---

- [ ] Skim through all files:
    - [ ] platform_sh/deploy.py
    - [ ] fly_io/deploy.py
    - [ ] heroku/deploy.py

**Comment by ehmatthes** [comment]
First pass at testing
---

Start with integration tests first, because they are most important and don't check log output. If those pass, unit tests should be mostly updating log assertions?

- [ ] integration tests
    - [ ] platform_sh (start here, because these almost always pass)
    - [ ] fly_io (next, because just ran these)
    - [ ] heroku (last, because least likely to pass based on staleness)
- [ ] unit tests
    - [ ] platform_sh
    - [ ] fly_io
    - [ ] heroku

**Comment by ehmatthes** [comment]
platform_sh integration test
---

- [ ] Hangs at login required.
    - [ ] Catch this, or otherwise surface it to user.

**Comment by ehmatthes** [comment]
fly_io integration test
---

- [ ] Need to update fly CLI.

**Comment by ehmatthes** [comment]
Update unit tests
---

- [ ] Look at *test_project_inspection.py*.

**Comment by ehmatthes** [comment]
build_dev_env.py cross-platform
---

- [ ] `subprocess.run()` needs to use `cmd`, not `cmd.parts()`.
- [ ] Address `-a` issue.
- [ ] Address line endings issue.
- [ ] Make sure integration tests still work on macOS.

**Comment by ehmatthes** [comment]
Platform.sh integration test (Windows)
---

- [ ] `git status` shows .platform/ directory.
    - [ ] Why is this being added before the git status check?
    - [ ] Is this happening on macOS as well?
    - [ ] Resolve on both platforms.

**Comment by ehmatthes** [comment]
Moving
